src/plugins/intel_gpu/src/graph/range.cpp

```diff
@@ -22,15 +24,49 @@ primitive_type_id range::type_id() {
 }
 
 layout range_inst::calc_output_layout(range_node const& node, kernel_impl_params const& impl_param) {
-    return impl_param.typed_desc<range>()->output_layout;
+    auto desc = impl_param.typed_desc<range>();
+    auto out_dtype = desc->output_data_type.value_or(impl_param.get_input_layout().data_type);
+    return {desc->output_pshape, out_dtype, cldnn::format::bfyx};
+}
+
+template<typename ShapeType>
+std::vector<layout> range_inst::calc_output_layouts(range_node const& /*node*/, kernel_impl_params const& impl_param) {
+    auto desc = impl_param.typed_desc<range>();
+    auto output_data_type = desc->output_data_type.value_or(impl_param.get_input_layout().data_type);
+
+    ov::op::v4::Range op;
+    op.set_output_type(data_type_to_element_type(output_data_type));
+    std::vector<ShapeType> output_shapes = {ShapeType::dynamic(1)};
+    std::vector<ShapeType> input_shapes = {ov::Shape(), ov::Shape(), ov::Shape()};
```

Comment:
Shouldn't these come from the impl_param.input_layouts? Why are they remain default constructor?

Reply:
range::shape_infer should have empty Shape() for each input shapes because each input should be scalar value.
If I add input shape from impl_params to the input_shapes, ngraph shape_infer will throw exception such as **"'start' input is not a scalar"**

Reply:
I will update it on new PR after this PR is merged as we discussed offline.